code/job.py

```diff
@@ -0,0 +1,53 @@
+from flask_restful import Resource, reqparse
+
+jobs = {}
+
+class Job(Resource):
+    parser = reqparse.RequestParser()
```

Comment:
This might be a perfectly valid Python thing, but how come the parser is being set up as what would be called a static member in C#? i.e. rather than putting it in the init method and assigning it to self.parser.

The main reason I'm asking is that this class feels difficult to write unit tests for. I don't know how much of a thing this is in the Flask world, but you can use this package:

https://github.com/alecthomas/injector

to add the ability to inject dependencies in to url handlers. So, here you could have the parser injected in the constructor, which would then give you the ability to mock it in a unit test, and therefore be able to write unit tests much more easily.

Reply:
To expand on this some more; I know that what you've got here is a very simple example ( no data store, for example), but I would be looking to split the data store (jobs) and the code that interacts with it, out in to a separate service class. This service class would then be injected in to this class. So this class would consist of the code to deal with getting the fields from the request, and the service class would deal with that data.

Does that make sense?